packages/peregrine/lib/talons/AuthModal/useAuthModal.js

```diff
@@ -0,0 +1,52 @@
+import { useCallback, useEffect, useState } from 'react';
+import { useUserContext } from '../../context/user';
+
+const UNAUTHED_ONLY = ['CREATE_ACCOUNT', 'FORGOT_PASSWORD', 'SIGN_IN'];
+
+export const useAuthModal = props => {
+    const {
+        closeDrawer,
+        showCreateAccount,
+        showForgotPassword,
+        showMainMenu,
+        showMyAccount,
+        view
+    } = props;
+
+    const [username, setUsername] = useState('');
+    const [{ currentUser }, { createAccount, signOut }] = useUserContext();
+
+    // if the user is authed, the only valid view is "MY_ACCOUNT"
+    useEffect(() => {
+        if (currentUser && currentUser.id && UNAUTHED_ONLY.includes(view)) {
```

Comment:
We probably don't even need that last conditional, right?
```suggestion
        if (currentUser && currentUser.id) {
```

Reply:
Sure. I can change this.

Reply:
Did we end up needing it after all?